provisioning_client/tests/hsm_client_tpm_ut/hsm_client_tpm_ut.c

```diff
@@ -127,6 +127,72 @@ static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
 
 static TEST_MUTEX_HANDLE g_testByTest;
 
+char* umock_stringify_TPMA_SESSION(const TPMA_SESSION* value)
+{
+    char* result = (char*)my_gballoc_malloc(1);
+    (void)value;
+    result[0] = '\0';
+    return result;
+}
+
+int umock_are_equal_TPMA_SESSION(const TPMA_SESSION* left, const TPMA_SESSION* right)
+{
+    int result;
+
+    if (((left == NULL) && (right != NULL)) || ((right == NULL) && (left != NULL)))
+    {
+        result = 0;
+    }
+    else
+    {
+        if (left == right)
```

Comment:
So our comparison is by pointer address only (shallow compare)? Is that enough?

Reply:
No its not enough, fixed